clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/TestVenicePushJobCheckpoints.java

```diff
@@ -61,85 +64,273 @@ public class TestVenicePushJobCheckpoints {
+            // ValidateSchemaAndBuildDictMapper related counters below
+            // Number of Processed files
             new MockCounterInfo(
                 MRJobCounterHelper.MAPPER_NUM_RECORDS_SUCCESSFULLY_PROCESSED_GROUP_COUNTER_NAME,
-                NUMBER_OF_FILES_TO_READ_AND_BUILD_DICT_COUNT + 1), // Number of Processed files in
-                                                                   // ValidateSchemaAndBuildDictMapper
-            new MockCounterInfo(MRJobCounterHelper.MAPPER_ZSTD_DICT_TRAIN_SUCCESS_GROUP_COUNTER_NAME, 1) // Dictionary
-                                                                                                         // building
-                                                                                                         // succeeded if
-                                                                                                         // enabled
-        ),
+                NUMBER_OF_FILES_TO_READ_AND_BUILD_DICT_COUNT + 1)),
         Arrays.asList(
             VenicePushJob.PushJobCheckpoints.INITIALIZE_PUSH_JOB,
+            VenicePushJob.PushJobCheckpoints.VALIDATE_SCHEMA_AND_BUILD_DICT_MAP_JOB_COMPLETED,
             VenicePushJob.PushJobCheckpoints.NEW_VERSION_CREATED,
-            VenicePushJob.PushJobCheckpoints.WRITE_ACL_FAILED),
+            VenicePushJob.PushJobCheckpoints.MAP_REDUCE_JOB_COMPLETED,
+            VenicePushJob.PushJobCheckpoints.JOB_STATUS_POLLING_COMPLETED),
         properties -> {
+          properties.setProperty(USE_MAPPER_TO_BUILD_DICTIONARY, "true");
           properties.setProperty(COMPRESSION_METRIC_COLLECTION_ENABLED, "false");
+        });
+  }
+
+  @Test
+  public void testWithCompressionCollectionDisabled() throws Exception {
+    testHandleErrorsInCounter(
+        Arrays.asList(
+            new MockCounterInfo(MRJobCounterHelper.TOTAL_VALUE_SIZE_GROUP_COUNTER_NAME, 1),
+            new MockCounterInfo(MRJobCounterHelper.WRITE_ACL_FAILURE_GROUP_COUNTER_NAME, 0),
+            new MockCounterInfo(MRJobCounterHelper.DUP_KEY_WITH_DISTINCT_VALUE_GROUP_COUNTER_NAME, 0),
+            // All reducers closed
+            new MockCounterInfo(MRJobCounterHelper.REDUCER_CLOSED_COUNT_GROUP_COUNTER_NAME, PARTITION_COUNT)),
+        Arrays.asList(
+            VenicePushJob.PushJobCheckpoints.INITIALIZE_PUSH_JOB,
+            VenicePushJob.PushJobCheckpoints.NEW_VERSION_CREATED,
+            VenicePushJob.PushJobCheckpoints.MAP_REDUCE_JOB_COMPLETED,
+            VenicePushJob.PushJobCheckpoints.JOB_STATUS_POLLING_COMPLETED),
+        properties -> {
           properties.setProperty(USE_MAPPER_TO_BUILD_DICTIONARY, "false");
+          properties.setProperty(COMPRESSION_METRIC_COLLECTION_ENABLED, "false");
         });
   }
 
-  @Test(expectedExceptions = { VeniceException.class })
-  public void testHandleDuplicatedKeyWithDistinctValue() throws Exception {
+  @Test
```

Comment:
Is the original expected exception being checked somewhere? Same to below a few test cases.

Reply:
These are different tests. 
Github diff view is showing these up side by side as this file is sort of completely edited as I made lot of comments to be single line rather than multi liners. 
But for the test your question was about, actually added `expectedExceptionsMessageRegExp` to make it more specific.